File: balanced.py
# Given an array of parentheses check if opening and closing elements are same quantity
open = ['[','{','(']
closed = [']','}',')']

def balanced(arr):
    balance = 0
    for a in arr:
        if a in open:
            balance += 1
        if a in closed:
            balance -= 1

    if balance == 0:
        return True

    return False

arr = ['[',']','[','[',']','}',')']
resp = balanced(arr)
print(resp)

File: balanced.js
function balanced(arr) {
  let open = 0;
  let closed = 0;
  for(let i=0; i<arr.length; i++) {
    if(arr[i] === '(' || arr[i] === '{' || arr[i] === '[')
      open++;
    if(arr[i] === ')' || arr[i] === '}' || arr[i] === ']')
      closed++;
  }
  if(closed === open)
    return true;
  else {
    return false;
  }
}

const arr = ['(','(','(',']',']',']'];
res = balanced(arr);
console.log(res);
